activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -910,14 +907,18 @@ def remove_connection(owner)
       # #fetch is significantly slower than #[]. So in the nil case, no caching will
       # take place, but that's ok since the nil case is not the common one that we wish
       # to optimise for.
```

Comment:
@arthurnn This comment mentions that `#fetch is significantly slower than #[]`, and previous versions were optimized for the common case, and not caching in the `nil` case.

Shouldn’t this version also use `#[]` instead of `#fetch`?